specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -234,65 +233,158 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
       ),
     };
 
-    const hasSameValues = (
+    const getFieldValue = async (
```

Comment:
are these functions using any variables from the scope in which they are defined (the `uniqueIn` method)?
if not, perhaps you can turn them into functions outside the class, or their own methods?
`uniqueIn` is kind of a hot-path function, so re-creating large arrow functions in here could hurt the performance (makes the job of the optimizer harder)